src/toil/jobStores/awsJobStore.py

```diff
@@ -865,3 +981,66 @@ def parse_bool(s):
         return False
     else:
         raise ValueError(s)
+
+class File(object):
+    """
+    >>> f = File(None, None)
+    >>> f.version=3
+    >>> f.version
+    3
+    >>> f.version=4
+    Traceback (most recent call last):
+    ...
+    AssertionError
+    >>> f.buffer={}
+    Traceback (most recent call last):
+    ...
+    AssertionError
+    >>> f = File(None, None)
+    >>> f.buffer={}
+    >>> f.buffer
+    {}
+    >>> f.buffer={}
+    Traceback (most recent call last):
+    ...
+    AssertionError
+    >>> f.version=4
+    Traceback (most recent call last):
+    ...
+    AssertionError
+    """
+    typeRegular="regular"
+    typeStats="stats"
+    def __init__(self, jobStore,jobStoreFileID, fileType):
+        """
+        :param fileType: "regular" or "stats"
+        """
+        self.jobStore=jobStore
+        self.jobStoreFileID=jobStoreFileID
+        self.fileType=fileType
+        self.jobStoreID=None
+        self._version=None
+        self._buffer=None
+
+    @property
+    def bucket(self):
+        if self.fileType==File.typeRegular:
+            return self.jobStore.files
+        elif self.fileType==File.typeStats:
+            return self.jobStore.stats
+        else:
+            assert False
+    @property
+    def version(self):
+        return self._version
+    @version.setter
+    def version(self, version):
+        assert self._version is None and self._buffer is None
+        self._version = version
+    @property
+    def buffer(self):
+        return self._buffer
+    @buffer.setter
+    def buffer(self, buffer):
+        assert self._buffer is None and self._version is None
+        self._buffer = buffer
```

Comment:
Please set you editor to ensure files end in a newline. I'm not being pedantic, missing newlines at EOF have caused me to waste hours of life tracking down bugs with the last line of a file not being read.